network/p2pNetwork.go

```diff
@@ -735,6 +851,13 @@ func (n *P2PNetwork) txTopicHandleLoop() {
 		// from gossipsub's point of view, it's just waiting to hear back from the validator,
 		// and txHandler does all its work in the validator, so we don't need to do anything here
 		_ = msg
+
+		// participation or configuration change, cancel subscription and quit
+		if !n.wantTXGossip.Load() {
```

Comment:
Hm, I'm not sure how often `OnNetworkAdvance` can be called where it changes `n.wantTXGossip`. If this runs less often than `OnNetworkAdvance`, it's possible this routine won't notice swaps to `n.wantTXGossip` and multiple routines could be listening. That could be addressed by storing the "config" version as an int and bailing out here if that version isn't what the routine was started with.

But in reality I'm assuming it will be very rare for `n.wantTXGossip` to change?